README.md

```diff
@@ -91,120 +124,235 @@ On a Skylake processor, the parsing speeds (in GB/s) of various processors on th
 
-If you are processing large files (e.g., 100 MB), it is possible that the performance of simdjson will be limited by page misses and/or page allocation. [On some systems, memory allocation runs far slower than we can parse (e.g., 1.4GB/s).](https://lemire.me/blog/2020/01/14/how-fast-can-you-allocate-a-large-block-of-memory-in-c/)
+  // Casting a JSON element to an integer
+  uint64_t year = car["year"];
+  cout << "- This car is " << 2020 - year << "years old." << endl;
 
-A viable strategy is to amortize the cost of page allocation by reusing the same `parser` object over several files:
+  // Iterating through an array of floats
+  double total_tire_pressure = 0;
+  for (double tire_pressure : car["tire_pressure"]) {
+    total_tire_pressure += tire_pressure;
+  }
+  cout << "- Average tire pressure: " << (total_tire_pressure / 4) << endl;
 
-```C++
-// create one parser
-simdjson::document::parser parser;
-...
-// the parser is going to pay a memory allocation price
-auto [doc1, error1] = parser.parse(largestring1);
-...
-// use again the same parser, it will be faster
-auto [doc2, error2] = parser.parse(largestring2);
-...
-auto [doc3, error3] = parser.load("largefilename"); 
+  // Writing out all the information about the car
+  for (auto [key, value] : car) {
+    cout << "- " << key << ": " << value << endl;
+  }
+}
 ```
 
-If you cannot reuse the same parser instance, maybe because your application just processes one large document once, you will get best performance with large or huge pages. Under Linux, you can enable transparent huge pages with a command like `echo always > /sys/kernel/mm/transparent_hugepage/enabled` (root access may be required). It may be more difficult to achieve the same result under other systems like macOS or Windows.
+### JSON Pointer
 
-In general, when running benchmarks over large files, we recommend that you report performance numbers with and without huge pages if possible. Furthermore, you should amortize the parsing (e.g., by parsing several large files) to distinguish the time spent parsing from the time spent allocating memory. 
+simdjson also supports JSON pointer, letting you reach further down into the document:
 
+```c++
+auto cars_json = R"( [
+  { "make": "Toyota", "model": "Camry",  "year": 2018, "tire_pressure": [ 40.1, 39.9, 37.7, 40.4 ] },
+  { "make": "Kia",    "model": "Soul",   "year": 2012, "tire_pressure": [ 30.1, 31.0, 28.6, 28.7 ] },
+  { "make": "Toyota", "model": "Tercel", "year": 1999, "tire_pressure": [ 29.8, 30.0, 30.2, 30.5 ] }
+] )"_padded;
+document::parser parser;
+document &cars = parser.parse(cars_json);
+cout << cars["/0/tire_pressure/1"] << endl; // Prints 39.9
+```
 
-## Including simdjson
+### Error Handling
 
+All simdjson APIs that can fail return `simdjson_result<T>`, which is a &lt;value, error_code&gt;
+pair. The error codes and values can be accessed directly, reading the error like so:
 
-## Code usage and example
+```c++
+auto [doc, error] = parser.parse(json);
```

Comment:
It might be useful, maybe as a comment in the pseudcode, to spell out the types of doc and error in this instance. What is error and what can I do with it... What is doc and what can I do with it?